lib/codec.js

```diff
@@ -368,4 +395,19 @@
     }
   }
 
+  var WrappedMessage = (function() {
```

Comment:
Can we move `WrappedMessage` inside `encodeWrappedMessage` since it's only used there?

Reply:
Yeah, but I think it will also be used in the decodeWrappedMessage function.

Reply:
Ok. Just a detail: I would have avoided the definition of `encode` and `decode` functions. Maybe returning `wrappedMessage` directly?

Reply:
Yes sir. That would be great imo also. I will change that.